sound/soc/sof/pm.c

```diff
@@ -258,6 +258,7 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 
 	/* reset FW state */
 	sdev->fw_state = SOF_FW_BOOT_NOT_STARTED;
+	sdev->enabled_cores_mask = 0;
```

Comment:
much simpler indeed. But just thinking out loud, you have core 0 and core 1 on before suspendng, does turning off Core 0 guarantee that core 1 is turned off too?

Reply:
I don't think this PR turns off anything, it just resets a mask so that the state is the same as what firmware expects on restart. it's a good question as to what happens if the core1..N are not explicitly powered-off, but that's not what the error was in issue #2824

Reply:
I removed `sdev->enabled_cores_mask = 0;` was because that I think we will set `sdev->enabled_cores_mask` if we call snd_sof_dsp_core_power_down() to power down a core. I am not sure if all cores will be powered down in suspend. Should we make sure that all cores are powered down before setting `sdev->enabled_cores_mask = 0;`?

Reply:
> I removed `sdev->enabled_cores_mask = 0;` was because that I think we will set `sdev->enabled_cores_mask` if we call snd_sof_dsp_core_power_down() to power down a core. I am not sure if all cores will be powered down in suspend. Should we make sure that all cores are powered down before setting `sdev->enabled_cores_mask = 0;`?

That's what I named it decent fix, IMHO we should really avoid changing 'sdev->enabled_cores_mask' directly, as much as we can. once this part is refined, with something like #2850, we should call a _put() instead in the _suspend().

Reply:
@kv2019i what's the scope of this PR? Is it intended as a "hot fix" until #2850 implements a proper DSP core ref-counting at which point this should become redundant again or is this supposed to eliminate the need for #2850 completely for now?

Reply:
@lyakh This is a hot-fix until #2850 is implemented and something we can send to 5.13 as a bugfix.

Reply:
@bardliao Ack, I did review your original patch and thought exactly the same, but alas I think we missed that in the suspend/resume cycles, there's no controlled unload of the topology. There Rather we power down the whole DSP, and start from scratch. If we do driver unload, then sof_widget_unload() is called and at least the pipe scheduler cores get powered down, but this won't help in suspend/resume (and won't help if dynamic pipelines are enabled).